routes/api.js

```diff
@@ -26,6 +28,20 @@ router.get('/merchantproducts', async (req, res) => {
   res.send(products);
 });
 
+router.get('/search/:query', async (req, res) => {
+  try {
+    const searchQuery = req.params.query;
+    const foundProducts = await Products.find({
+      title: { $regex: '.*' + searchQuery + '.*', $options: 'i' }
```

Comment:
For later optimizations, you should create a text index for `title`:
https://docs.mongodb.com/manual/core/index-text/